### PR TITLE
2023 10 04 added alarm state points

### PR DESCRIPTION
**Pull request by wroantree-willow**

### CODE REVIEW

File: Ontology/Willow/Capability/State/Binary State/Alarm State/Overload/CurrentOverloadAlarmState.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "@id": "dtmi:com:willowinc:CurrentOverloadAlarmState;1",
```
Comment (rszcodronski) [inline Ontology/Willow/Capability/State/Binary State/Alarm State/Overload/CurrentOverloadAlarmState.json:2]: Is this different than the other Current Alarm States such as HighCurrentAlarmState?

File: Ontology/Willow/Capability/State/Binary State/Alarm State/Pressure/StaticPressure/Refrigerant/LowCompressorEnteringRefrigerantStaticPressureAlarmState.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "@id": "dtmi:com:willowinc:LowCompressorEnteringRefrigerantStaticPressureAlarmState;1",
+  "@type": "Interface",
+  "displayName": {
+    "en": "Low Compressor Entering Refrigerant Static Pressure Alarm State"
+  },
+  "description": {"en": "The Boolean state of the low compressor entering refrigerant static pressure alarm is 'alarm' when 'true' and 'normal' when 'false'"},
```
Comment (rszcodronski) [inline Ontology/Willow/Capability/State/Binary State/Alarm State/Pressure/StaticPressure/Refrigerant/LowCompressorEnteringRefrigerantStaticPressureAlarmState.json:7]: The order of terms in the id, display name, and description seem jumbled from how you would likely say this example.

File: Ontology/Willow/Capability/Parameter/Limit/HighLimit.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
-  "@id": "dtmi:com:willowinc:MaxLimit;1",
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/Parameter/Limit/HighLimit.json:2]: In talking to Mapped, Min/Max are more akin to physical limits whereas High/Low are control limits. We can retain min and max for continuity but we should include high/low and have them be the parent models for the child points

File: Ontology/Willow/Capability/Parameter/Limit/LowLimit.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
-  "@id": "dtmi:com:willowinc:MinLimit;1",
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/Parameter/Limit/LowLimit.json:2]: In talking to Mapped, Min/Max are more akin to physical limits whereas High/Low are control limits. We can retain min and max for continuity but we should include high/low and have them be the parent models for the child points

File: Ontology/Willow/Capability/Parameter/Limit/Quantity/Frequency/FrequencyHighLimit.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+  "@id": "dtmi:com:willowinc:FrequencyHighLimit;1",
+  "@type": "Interface",
+  "displayName": {
+    "en": "Frequency High Limit"
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/Parameter/Limit/Quantity/Frequency/FrequencyHighLimit.json:5]: Neither, which is why it's slightly different. The child models that required this parent model are the VFD Frequency High and Low Limits. They're limits programmed into the VFD such that the 0-100% speed signal that the VFD receives will span from the VFD Frequency Low Limit, let's say 20 Hz, to the VFD Frequency High Limit, typically 60Hz.

File: Ontology/Willow/Capability/Sensor/QuantitySensor/CountSensor/StageCountSensor/StageCountSensor.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+  "@id": "dtmi:com:willowinc:StageCountSensor;1",
+  "@type": "Interface",
+  "displayName": {
+    "en": "Stage Count Sensor"
+  },
+  "description": {"en": "The counted number of active stages"},
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/Sensor/QuantitySensor/CountSensor/StageCountSensor/StageCountSensor.json:7]: Sure, I'll clarify

File: Ontology/Willow/Capability/Sensor/QuantitySensor/CountSensor/StageCountSensor/CellStageCountSensor.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{
+  "@id": "dtmi:com:willowinc:CellStageCountSensor;1",
+  "@type": "Interface",
+  "displayName": {
+    "en": "Cell Stage Count Sensor"
+  },
+  "description": {"en": "The counted number of active cells"},
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/Sensor/QuantitySensor/CountSensor/StageCountSensor/CellStageCountSensor.json:7]: It refers to the active quantity of cells of a cooling tower. I'll clarify.

File: Ontology/Willow/Capability/State/Binary State/Alarm State/CondensateHighLevelAlarmState.json [hunk 8]
```diff
@@ -1,16 +0,0 @@
-{
-  "@id": "dtmi:com:willowinc:CondensateHighLevelAlarmState;1",
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/State/Binary State/Alarm State/CondensateHighLevelAlarmState.json:2]: No deprecation should be required, it was added to the ontology very recently. It was replaced by "dtmi:com:willowinc:HighCondensateLevelAlarmState;1" under the newly created Level/Water Level alarms

File: Ontology/Willow/Capability/State/Binary State/Alarm State/Emergency/EmergencyStopAlarmState.json [hunk 9]
```diff
@@ -0,0 +1,17 @@
+{
+  "@id": "dtmi:com:willowinc:EmergencyStopAlarmState;1",
+  "@type": "Interface",
+  "displayName": {
+    "en": "Emergency Stop Alarm State"
+  },
+  "description": {"en": "The Boolean state of the emergency stop alarm is 'alarm' when 'true' and 'normal' when 'false'"},
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/State/Binary State/Alarm State/Emergency/EmergencyStopAlarmState.json:7]: An Emergency Stop is a manually actuated mechanism that typically triggers a relay to sever power to a piece of equipment.

File: Ontology/Willow/Capability/State/Binary State/Alarm State/Failure/UnitFailureAlarmState.json [hunk 10]
```diff
@@ -0,0 +1,17 @@
+{
+  "@id": "dtmi:com:willowinc:UnitFailureAlarmState;1",
+  "@type": "Interface",
+  "displayName": {
+    "en": "Unit Failure Alarm State"
```
[the remark this answers is not in the document]
Comment (wroantree-willow) [reply Ontology/Willow/Capability/State/Binary State/Alarm State/Failure/UnitFailureAlarmState.json:5]: A Unit Failure alarm is as you stated, when the entire piece of equipment has failed. Unit Failure alarms typically come from packaged controllers and are labeled as Unit Failure alarms. I can change to Equipment Failure alarm to broaden the applications.

File: Ontology/Willow/Capability/State/Binary State/Alarm State/Overload/CurrentOverloadAlarmState.json (hunk 1 above)
Comment (wroantree-willow) [reply Ontology/Willow/Capability/State/Binary State/Alarm State/Overload/CurrentOverloadAlarmState.json:2]: Yes, the two models are borne of different devices. For the HighCurrentAlarmState model, a current sensor measures the current and another device compares that to an alarm threshold that's typically adjustable. CurrentOverloadAlarmState would come from a circuit breaker or similar device that has an alarm contact to indicate when it has tripped.

**Review by rszcodronski — APPROVED** [review]

**Pull request merged**